server/api/ballot_manifest.py

```diff
@@ -228,12 +231,25 @@ def process() -> None:
         session.commit()
 
 
+def is_batch_inventory_worksheet(first_line: bytes) -> bool:
+    return first_line.decode("utf-8").strip() == "Batch Inventory Worksheet"
+
+
+def validate_is_not_batch_inventory_worksheet(file: FileStorage):
+    first_line = file.stream.readline()
+    file.stream.seek(0)
+    if is_batch_inventory_worksheet(first_line):
+        raise BadRequest(BATCH_INVENTORY_WORKSHEET_UPLOADED_ERROR)
+
+
 # Raises if invalid
 def validate_ballot_manifest_upload(request: Request):
     if "manifest" not in request.files:
         raise BadRequest("Missing required file parameter 'manifest'")
 
-    validate_csv_mimetype(request.files["manifest"])
+    file = request.files["manifest"]
+    validate_is_not_batch_inventory_worksheet(file)
+    validate_csv_mimetype(file)
```

Comment:
> Note that because CSV validation currently happens in upload before process(), it's easiest to trigger a toast notification rather than permanent job error under the upload field.

This doesn't match my reading. I see CSV mimetype validation happening during the initial file upload, but the actual parsing of the CSV file seems to happen within the background task.

If possible, making this a file processing error would be best since it will be persistent in the UI.

Reply:
You are absolutely right, sorry! Updated